packages/fxa-settings/src/components/UnitRowTwoStepAuth/index.tsx

```diff
@@ -0,0 +1,126 @@
+import AlertBar from '../AlertBar';
+import Modal from '../Modal';
+import UnitRow from '../UnitRow';
+import VerifiedSessionGuard from '../VerifiedSessionGuard';
+import { useAccount } from '../../models';
+
+export const DESTROY_TOTP_TOKEN_MUTATION = gql`
+  mutation destroyTotpToken($input: DestroyTotpToken!) {
+    destroyTotpToken(input: $input) {
+      clientMutationId
+    }
+  }
+`;
+
+export const UnitRowTwoStepAuth = () => {
+  const { totp } = useAccount();
+  const { exists, verified } = totp;
+  const [modalRevealed, revealModal, hideModal] = useBooleanState();
+  const [alertBarRevealed, revealAlertBar, hideAlertBar] = useBooleanState();
+  const [errorText, setErrorText] = useState<string>();
+  const onError = (e: Error) => {
+    setErrorText(e.message);
+    hideModal();
+    revealAlertBar();
+  };
+  const [disableTwoStepAuth] = useMutation(DESTROY_TOTP_TOKEN_MUTATION, {
+    variables: { input: {} },
+    onCompleted: () => {
+      hideModal();
+      revealAlertBar();
+    },
+    onError,
+    ignoreResults: true,
+    update: (cache) => {
+      cache.modify({
+        fields: {
+          account: (existing) => {
+            return { ...existing, totp: { exists: false, verified: false } };
+          },
+        },
+      });
+    },
+  });
+
+  const conditionalUnitRowProps = exists
+    ? {
+        headerValueClassName: 'text-green-800',
+        headerValue: 'Enabled',
+        ctaText: 'Change',
+        secondaryCtaText: 'Disable',
+        secondaryButtonClassName: 'cta-caution',
+        revealSecondaryModal: revealModal,
+      }
+    : {
+        headerValue: null,
+        noHeaderValueText: 'Not Set',
+        ctaText: 'Add',
+        secondaryCtaText: undefined,
+        revealSecondaryModal: undefined,
+      };
+
+  return (
+    <UnitRow
+      header="Two-step authentication"
+      route="/beta/settings/two_step_authentication"
+      {...conditionalUnitRowProps}
+    >
+      <p className="text-sm mt-3">
+        Prevent someone else from logging in by requiring a unique code only you
+        have access to.
+      </p>
+      {modalRevealed && (
+        <VerifiedSessionGuard onDismiss={hideModal} onError={onError}>
+          <Modal
+            onDismiss={hideModal}
+            onConfirm={disableTwoStepAuth}
+            headerId="two-step-auth-disable-header"
+            descId="two-step-auth-disable-description"
+            confirmText="Disable"
+            confirmBtnClassName="cta-caution"
+          >
+            <h2 className="font-bold text-xl text-center mb-2">
+              Disable Two-step authentication?
+            </h2>
+            {/* "replacing recovery codes" link below will actually drop you into
+            recovery codes flow in the future. */}
+            <p>
+              You won't be able to undo this action. You also have the option of
+              <LinkExternal
+                className="link-blue text-xs mt-2"
```

Comment:
This class makes the link awkward. You can remove `text-sm mt-2`. Additionally I think you'll need to add `{' '}` right before the opening LinkExternal tag to force a space.
![Screen Shot 2020-08-27 at 2 40 22 PM](https://user-images.githubusercontent.com/6392049/91481968-484b5100-e873-11ea-84de-98ff92285baf.png)